mjml/__init__.pyi

```diff
@@ -0,0 +1,35 @@
+import pathlib
+import typing as t
+
+if t.TYPE_CHECKING:
+    import typing_extensions as te
+    from _typeshed import SupportsRead
+
+    from mjml.core.api import Component
+
+    class _Output(t.NamedTuple):
+        html: str
+        errors: t.Sequence[str]
+
+        @te.overload
+        def __getitem__(self, _: t.Literal["html"]) -> str: ...
+        @te.overload
+        def __getitem__(self, _: t.Literal["errors"]) -> t.Sequence[str]: ...
+        @te.overload
+        def get(self, key: t.Literal["html"], /) -> t.Optional[str]: ...
+        @te.overload
+        def get(self, key: t.Literal["html"], default: str, /) -> str: ...
+        @te.overload
+        def get(self, key: t.Literal["errors"], /) -> t.Optional[t.Sequence[str]]: ...
+        @te.overload
+        def get(self, key: t.Literal["errors"], default: t.Sequence[str], /) -> t.Sequence[str]: ...
+
+    FpOrJson = t.Union[t.Dict[str, t.Any], str, bytes, SupportsRead[str], SupportsRead[bytes]]
```

Comment:
Probably bike-shedding but does `mjml_to_html()` works if `xml_fp_or_json` contains a `bytes` value?

Reply:
The `BeatifulSoup` constructor has the following annotation (in the stubs):

```python
def __init__(
        self,
        markup: str | bytes | SupportsRead[str] | SupportsRead[bytes] = "",
       # ...
```

Purely based on that it *should* work, since `bytes` is left untouched by `mjml_to_html`, but I also quickly tried it, with a minimal example, and it seems to work just fine:

```python
from mjml import mjml_to_html

mjml_to_html(b"<mjml><mj-body><mj-raw>foo</mj-raw></mj-body></mjml>")

# DotMap(html='<!doctype html>\n<html xmlns="http://www.w3.org/1999/xhtml" xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office">\n  <head>\n    <title>\n      \n    </title>\n    <!--[if !mso]><!-->\n    <meta http-equiv="X-UA-Compatible" content="IE=edge">\n    <!--<![endif]-->\n    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">\n    <meta name="viewport" content="width=device-width, initial-scale=1">\n    <style type="text/css">\n      #outlook a { padding:0; }\n      body { margin:0;padding:0;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%; }\n      table, td { border-collapse:collapse;mso-table-lspace:0pt;mso-table-rspace:0pt; }\n      img { border:0;height:auto;line-height:100%; outline:none;text-decoration:none;-ms-interpolation-mode:bicubic; }\n      p { display:block;margin:13px 0; }\n    </style>\n    <!--[if mso]>\n    <noscript>\n    <xml>\n    <o:OfficeDocumentSettings>\n      <o:AllowPNG/>\n      <o:PixelsPerInch>96</o:PixelsPerInch>\n    </o:OfficeDocumentSettings>\n    </xml>\n    </noscript>\n    <![endif]-->\n    <!--[if lte mso 11]>\n    <style type="text/css">\n      .mj-outlook-group-fix { width:100% !important; }\n    </style>\n    <![endif]-->\n    \n    \n    <style type="text/css">\n\n    \n    \n    </style>\n    <style type="text/css"></style>\n    \n  </head>\n  <body style="word-spacing:normal;">\n    \n    <div style="">foo</div>\n  </body>\n</html>', errors=[])

```